### PR TITLE
installer: Check if MySQL user exists before creating it

### PR DESCRIPTION
**Pull request by mtrojan-ub**

### CODE REVIEW

File: cpp/installer.cc [hunk 1]
```diff
@@ -155,6 +155,20 @@ TemporaryChDir::~TemporaryChDir() {
 }
 
 
+void GitActivateCustomHooks(const std::string &repository) {
+    const std::string original_git_directory(repository + "/.git");
+    const std::string original_hooks_directory(original_git_directory + "/hooks");
+    const std::string custom_hooks_directory(repository + "/git-config/hooks");
+
+    if (FileUtil::IsDirectory(custom_hooks_directory) && FileUtil::IsDirectory(original_hooks_directory)) {
```
Comment (ruschein) [inline cpp/installer.cc]: and

File: cpp/lib/include/DbConnection.h [hunk 2]
```diff
@@ -141,6 +141,8 @@ class DbConnection {
     }
 
     void mySQLSyncMultipleResults();
+
+    bool mySQLUserExists(const std::string &user, const std::string host);
```
Comment (ruschein) [inline cpp/lib/include/DbConnection.h]: host should be a reference

File: cpp/lib/src/DbConnection.cc [hunk 3]
```diff
@@ -505,3 +505,11 @@ void DbConnection::MySQLImportFile(const std::string &sql_file, const std::strin
     db_connection.queryOrDie(sql_data);
     db_connection.mySQLSyncMultipleResults();
 }
+
+
+bool DbConnection::mySQLUserExists(const std::string &user, const std::string host) {
+    queryOrDie("SELECT COUNT(*) as user_count FROM mysql.user WHERE User='" + user + "' AND Host='" + host + "';");
+    DbResultSet result_set(getLastResultSet());
+    const DbRow result_row = result_set.getNextRow();
```
Comment (ruschein) [inline cpp/lib/src/DbConnection.cc]: please use parentheses here.

File: git-config/hooks/post-merge [hunk 4]
```diff
@@ -0,0 +1,19 @@
+#!/bin/bash
+#
+# Git hook for updating TueFind configuration
+# (only if TueFind is already installed)
+#
+CHANGED_FILES="$(git diff-tree -r --name-only --no-commit-id ORIG_HEAD HEAD)"
```
Comment (ruschein) [inline git-config/hooks/post-merge]: is there a lon-form version of -r?

Comment (mtrojan-ub) [reply git-config/hooks/post-merge]: unfortunately not

File: cpp/installer.cc (hunk 1 above)
Comment (mtrojan-ub) [reply cpp/installer.cc]: done

File: cpp/lib/src/DbConnection.cc (hunk 3 above)
Comment (mtrojan-ub) [reply cpp/lib/src/DbConnection.cc]: done

**Pull request merged**